### PR TITLE
Feature/clingen

### PR DESCRIPTION
**Pull request by rajatshuvro**

### CODE REVIEW

File: docs/data-sources/clingen.mdx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+import CLINGENJSON from './clingen-json.md'
+import ClinGenDosageJson from './clingen-dosage-json.md'
+import ClinGenGeneValidity from './clingen-gene-validity-json.md'
+
+## Overview
+
+ClinGen is a National Institutes of Health (NIH)-funded resource dedicated to building a central resource that defines the clinical relevance of genes and variants for use in precision medicine and research.
+
+:::info Publication
```
Comment (MichaelStromberg) [inline docs/data-sources/clingen.mdx:13]: The publication doesn't seem right: https://www.nejm.org/doi/full/10.1056/NEJMsr1406261

Ideally, we would try to find a publication for each aspect (i.e. gene-disease validity, dosage sensitivity map, etc.)

File: docs/data-sources/clingen.mdx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+#### Parsing
+We parse the ClinGen tsv file and extract the following:
+* chrom
+* chromStart (note this a 0-based coordinate)
+* chromEnd
+* attrTags
+* attrVals
+
+*attrTags* and *attrVals* are comma separated lists. *attrTags* contains the field keys and *attrVals* contains the field values. We will parse the following keys from the two fields:
```
Comment (MichaelStromberg) [inline docs/data-sources/clingen.mdx]: When mentioning fields, use the inline code notation (i.e. attrTags surrounded by backticks)

File: docs/data-sources/clingen.mdx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+| Rating | Possible Clinical Interpretation                                                        |
+|--------|-----------------------------------------------------------------------------------------|
+| 0      | No evidence to suggest that dosage sensitivity is associated with clinical phenotype    |
+| 1      | Little evidence suggesting dosage sensitivity is associated with clinical phenotype     |
+| 2      | Emerging evidence suggesting dosage sensitivity is associated with clinical phenotype   |
+| 3      | Sufficient evidence suggesting dosage sensitivity is associated with clinical phenotype |
+| 30     | Gene associated with autosomal recessive phenotype                                      |
+| 40     | Dosage sensitivity unlikely                                                             |
+
+Reference: [https://www.ncbi.nlm.nih.gov/projects/dbvar/clingen/help.shtml](https://www.ncbi.nlm.nih.gov/projects/dbvar/clingen/help.shtml)
```
Comment (MichaelStromberg) [inline docs/data-sources/clingen.mdx:131]: This was a nice addition!

File: docs/data-sources/clingen.mdx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+*attrTags* and *attrVals* are comma separated lists. *attrTags* contains the field keys and *attrVals* contains the field values. We will parse the following keys from the two fields:
+* parent (this will be used as the ID in our JSON output)
+* clinical_int
+* validated
+* phenotype (this should be a string array)
+* phenotype_id (this should be a string array)
+
+Observed losses and observed gains will be calculated from entries that share a common parent ID.
+* variants with a common parent ID and same coordinates are grouped
+    * create hashsets for phenotypes and phentypeIds
```
Comment (MichaelStromberg) [inline docs/data-sources/clingen.mdx]: Do we need to talk about hashsets in public documentation?

File: docs/data-sources/clingen.mdx [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    * calculated observed losses, observed gains
+    * Clinical siginificance and validation status are collapesed using the priority strategy described below
```
Comment (MichaelStromberg) [inline docs/data-sources/clingen.mdx]: `Significance` is misspelled. As is `collapsed`

File: docs/data-sources/clingen.mdx [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+* Variants with the same parent ID can have different coordinates (mapped to hg38)
+    * nsv491508 : chr14:105583663-106881350 and chr14:105605043-106766076 (only one example)
+    * we kept both variants
+
+## Conflict Resolution
+### Clinical siginificance priority
```
Comment (MichaelStromberg) [inline docs/data-sources/clingen.mdx]: Significance is misspelled

File: docs/data-sources/clingen.mdx (hunk 4 above)
Comment (MichaelStromberg) [reply docs/data-sources/clingen.mdx]: Also did you mean phenotypeIds?

File: docs/data-sources/clingen.mdx [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+GENE SYMBOL,GENE ID (HGNC),DISEASE LABEL,DISEASE ID (MONDO),SOP,CLASSIFICATION,ONLINE REPORT,CLASSIFICATION DATE
++++++++++++,++++++++++++++,+++++++++++++,++++++++++++++++++,+++++++++,++++++++++++++,+++++++++++++,+++++++++++++++++++
+A2ML1,HGNC:23336,Noonan syndrome with multiple lentigines,MONDO_0007893,SOP5,No Reported Evidence,https://search.clinicalgenome.org/kb/gene-validity/59b87033-dd91-4f1e-aec1-c9b1f5124b16--2018-06-07T14:37:47,2018-06-07T14:37:47.175Z
+A2ML1,HGNC:23336,cardiofaciocutaneous syndrome,MONDO_0015280,SOP5,No Reported Evidence,https://search.clinicalgenome.org/kb/gene-validity/fc3c41d8-8497-489b-a350-c9e30016bc6a--2018-06-07T14:31:03,2018-06-07T14:31:03.696Z
+A2ML1,HGNC:23336,Costello syndrome,MONDO_0009026,SOP5,No Reported Evidence,https://search.clinicalgenome.org/kb/gene-validity/ea72ba8d-cf62-44bc-86be-da64e3848eba--2018-06-07T14:34:05,2018-06-07T14:34:05.324Z
+```
+### Download URL
+[https://search.clinicalgenome.org/kb/gene-validity.csv](https://search.clinicalgenome.org/kb/gene-validity.csv)
+
+### Conlict Resolution
```
Comment (MichaelStromberg) [inline docs/data-sources/clingen.mdx]: `Conflict` is misspelled

**Review by MichaelStromberg — CHANGES_REQUESTED** [review]

File: docs/data-sources/clingen.mdx (hunk 1 above)
Comment (rajatshuvro) [reply docs/data-sources/clingen.mdx:13]: Thanks. Updated.

File: docs/data-sources/clingen.mdx (hunk 4 above)
Comment (rajatshuvro) [reply docs/data-sources/clingen.mdx]: removed line about hash set, fixed spelling.

File: docs/data-sources/clingen.mdx (hunk 7 above)
Comment (rajatshuvro) [reply docs/data-sources/clingen.mdx]: spelling corrected

**Review by MichaelStromberg — APPROVED** [review]

**Pull request merged**